authentication/managing_cloud_provider_credentials/cco-mode-sts.adoc

```diff
@@ -0,0 +1,744 @@
++
+[source,terminal]
+----
+$ aws iam create-open-id-connect-provider --url $OPENID_BUCKET_URL --thumbprint-list $BUCKET_FINGERPRINT --client-id-list openshift
+----
+
+.. Retain the returned ARN of the newly created identity provider. This ARN is later referred to as `<aws-iam-openid-arn>`.
+
+. Generate IAM roles:
+
+.. Locate all `CredentialsRequest` CRs in this release image that target the cloud you are deploying on:
++
+[source,terminal]
+----
+$ oc adm release extract quay.io/openshift-release-dev/ocp-release:4.<y>.<z>-x86_64 --credentials-requests --cloud=aws
+----
++
+Where `<y>` and `<z>` are the numbers corresponding to the version of {product-title} you are installing.
+
+.. For each `CredentialsRequest` CR, create an IAM role that grants the necessary permissions and establishes a trust relationship that trusts the  identity provider previously created.
++
+For example, for the openshift-machine-api-operator `CredentialsRequest` CR in `0000_30_machine-api-operator_00_credentials-request.yaml`, create an IAM role that allows an identity from the created OIDC provider created for the cluster, similar to the following:
++
+[source,json]
+----
+{
+    "Role": {
+        "Path": "/",
+        "RoleName": "openshift-machine-api-aws-cloud-credentials",
+        "RoleId": "ARSOMEROLEID",
+        "Arn": "arn:aws:iam::123456789012:role/openshift-machine-api-aws-cloud-credentials",
+        "CreateDate": "2021-01-06T15:54:13Z",
+        "AssumeRolePolicyDocument": {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {
+                        "Federated": "<aws-iam-openid-arn>"
+                    },
+                    "Action": "sts:AssumeRoleWithWebIdentity",
+                    "Condition": {
+                        "StringEquals": {
+                            "s3.<aws-region>.amazonaws.com/$BUCKET_NAME:aud": "openshift"
+                        }
+                    }
+                }
+            ]
+        },
+        "Description": "OpenShift role for openshift-machine-api/aws-cloud-credentials",
+        "MaxSessionDuration": 3600,
+        "RoleLastUsed": {
+            "LastUsedDate": "2021-02-03T02:51:24Z",
+            "Region": "<aws-region>"
+        }
+    }
+}
+----
++
+Where `<aws-iam-openid-arn>` is the returned ARN of the newly created identity provider.
+
+.. To further restrict the role such that only specific cluster `ServiceAccount` objects can assume the role, modify the trust relationship of each role by updating the `.Role.AssumeRolePolicyDocument.Statement[].Condition` field to the specific `ServiceAccount` object for each component.
+
+*** Modify the trust relationship of the `cluster-image-registry-operator` role to have the following condition:
++
+[source,json]
+----
+"Condition": {
+  "StringEquals": {
+    "s3.us-east-1.amazonaws.com/<aws-infra-name>-installer:sub": [
```

Comment:
@jeana-redhat  This line changing to `"s3.<aws-region>.amazonaws.com/$BUCKET_NAME:sub": [` would be better to make context consistent with line 482.